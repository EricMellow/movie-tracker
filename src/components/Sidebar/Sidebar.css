.sidebar {
  width: 31%;
}

h1 {
  font-size: 1rem;
  margin-bottom: 25px;
  margin-top: 0;
  text-transform: uppercase;
}

.sidebarCards {
  border-radius: 5px;
  box-shadow: inset 0px 9px 8px -5px #101010, inset 0px -9px 8px -5px #101010;
  height: 340px;
  overflow-y: scroll;
}