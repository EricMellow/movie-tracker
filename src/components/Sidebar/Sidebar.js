import React, { Component } from 'react';
import './Sidebar.css';
import { connect } from 'react-redux';
import Card from '../Card/Card';

export class Sidebar extends Component {

  render() {
    const currentPage = this.props.renderRecent ? this.props.recentMovies : this.props.favoriteMovies;
    const movieCards = currentPage.map(movie => {
      const movieMatch = this.props.recentMovies.find( recentMovie => {
        return recentMovie.movie_id === movie.movie_id
      })
      const backdrop = movieMatch.backdrop;
     
      return (
        <Card 
          key={movie.movie_id}
          title={movie.title} 
          backdrop={backdrop}
          rating={movie.vote_average}
          id={movie.movie_id}
        />
      );
    });

    return (
      <div className="sidebar">
        <h1>Recent Movies</h1>
        { movieCards }
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  recentMovies: state.recentMovies,
  favoriteMovies: state.favoriteMovies,
  renderRecent: state.renderRecent
});

export default connect(mapStateToProps)(Sidebar);

