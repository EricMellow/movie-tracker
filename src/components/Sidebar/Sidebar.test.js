import React from 'react';
import { Sidebar, mapStateToProps } from "./Sidebar";
import { shallow } from 'enzyme';

describe('Card', () => {
  let wrapper;
  let mockLocation;
  let mockRecentMovies;
  let mockFavoriteMovies;

  beforeEach(() => {
    mockLocation = { pathname: '/' };
    mockRecentMovies = [{ title: 'It Takes Two', movie_id: 344911 }, { title: 'Step Brothers', movie_id: 324982 }];
    mockFavoriteMovies = [{ title: 'Step Brothers', movie_id: 324982 }];

    wrapper = shallow(<Sidebar location={mockLocation} renderRecent={true} recentMovies={mockRecentMovies} />);
  });

  it('should match the snapshot if renderRecent is true', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if renderRecent is false', () => {
    mockLocation = { pathname: '/Favorite Movies' };

    wrapper = shallow(<Sidebar location={mockLocation} renderRecent={false} favoriteMovies={mockFavoriteMovies} recentMovies={mockRecentMovies} />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {

    it('should map recentMovies, favoriteMovies, and renderRecent to props', () => {
      const mockState = {
        recentMovies: [{ title: 'Fight Club' }],
        favoriteMovies: [],
        renderRecent: true,
        selectedMovie: 12345,
        userID: 13
      };
      const expected = {
        recentMovies: [{ title: 'Fight Club' }],
        favoriteMovies: [],
        renderRecent: true
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

});