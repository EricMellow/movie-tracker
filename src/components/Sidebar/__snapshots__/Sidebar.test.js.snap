// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot if renderRecent is false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar
    favoriteMovies={
        Array [
            Object {
              "movie_id": 324982,
              "title": "Step Brothers",
            },
          ]
    }
    location={
        Object {
            "pathname": "/Favorite Movies",
          }
    }
    recentMovies={
        Array [
            Object {
              "movie_id": 344911,
              "title": "It Takes Two",
            },
            Object {
              "movie_id": 324982,
              "title": "Step Brothers",
            },
          ]
    }
    renderRecent={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Favorite Movies
</h1>,
        <div
          className="sidebarCards"
>
          <div
                    className="cardContainer"
          >
                    <Connect(Card)
                              backdrop={undefined}
                              id={324982}
                              rating={undefined}
                              title="Step Brothers"
                    />
                    <hr />
          </div>
</div>,
      ],
      "className": "sidebar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorite Movies",
        },
        "ref": null,
        "rendered": "Favorite Movies",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="cardContainer"
>
              <Connect(Card)
                            backdrop={undefined}
                            id={324982}
                            rating={undefined}
                            title="Step Brothers"
              />
              <hr />
</div>,
          ],
          "className": "sidebarCards",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "324982",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Connect(Card)
                  backdrop={undefined}
                  id={324982}
                  rating={undefined}
                  title="Step Brothers"
/>,
                <hr />,
              ],
              "className": "cardContainer",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "backdrop": undefined,
                  "id": 324982,
                  "rating": undefined,
                  "title": "Step Brothers",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "hr",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Favorite Movies
</h1>,
          <div
            className="sidebarCards"
>
            <div
                        className="cardContainer"
            >
                        <Connect(Card)
                                    backdrop={undefined}
                                    id={324982}
                                    rating={undefined}
                                    title="Step Brothers"
                        />
                        <hr />
            </div>
</div>,
        ],
        "className": "sidebar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorite Movies",
          },
          "ref": null,
          "rendered": "Favorite Movies",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="cardContainer"
>
                <Connect(Card)
                                backdrop={undefined}
                                id={324982}
                                rating={undefined}
                                title="Step Brothers"
                />
                <hr />
</div>,
            ],
            "className": "sidebarCards",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "324982",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Connect(Card)
                    backdrop={undefined}
                    id={324982}
                    rating={undefined}
                    title="Step Brothers"
/>,
                  <hr />,
                ],
                "className": "cardContainer",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "backdrop": undefined,
                    "id": 324982,
                    "rating": undefined,
                    "title": "Step Brothers",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "hr",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card should match the snapshot if renderRecent is true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar
    location={
        Object {
            "pathname": "/",
          }
    }
    recentMovies={
        Array [
            Object {
              "movie_id": 344911,
              "title": "It Takes Two",
            },
            Object {
              "movie_id": 324982,
              "title": "Step Brothers",
            },
          ]
    }
    renderRecent={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Recent Movies
</h1>,
        <div
          className="sidebarCards"
>
          <div
                    className="cardContainer"
          >
                    <Connect(Card)
                              backdrop={undefined}
                              id={344911}
                              rating={undefined}
                              title="It Takes Two"
                    />
                    <hr />
          </div>
          <div
                    className="cardContainer"
          >
                    <Connect(Card)
                              backdrop={undefined}
                              id={324982}
                              rating={undefined}
                              title="Step Brothers"
                    />
                    <hr />
          </div>
</div>,
      ],
      "className": "sidebar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Recent Movies",
        },
        "ref": null,
        "rendered": "Recent Movies",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="cardContainer"
>
              <Connect(Card)
                            backdrop={undefined}
                            id={344911}
                            rating={undefined}
                            title="It Takes Two"
              />
              <hr />
</div>,
            <div
              className="cardContainer"
>
              <Connect(Card)
                            backdrop={undefined}
                            id={324982}
                            rating={undefined}
                            title="Step Brothers"
              />
              <hr />
</div>,
          ],
          "className": "sidebarCards",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "344911",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Connect(Card)
                  backdrop={undefined}
                  id={344911}
                  rating={undefined}
                  title="It Takes Two"
/>,
                <hr />,
              ],
              "className": "cardContainer",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "backdrop": undefined,
                  "id": 344911,
                  "rating": undefined,
                  "title": "It Takes Two",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "hr",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "324982",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Connect(Card)
                  backdrop={undefined}
                  id={324982}
                  rating={undefined}
                  title="Step Brothers"
/>,
                <hr />,
              ],
              "className": "cardContainer",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "backdrop": undefined,
                  "id": 324982,
                  "rating": undefined,
                  "title": "Step Brothers",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "hr",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Recent Movies
</h1>,
          <div
            className="sidebarCards"
>
            <div
                        className="cardContainer"
            >
                        <Connect(Card)
                                    backdrop={undefined}
                                    id={344911}
                                    rating={undefined}
                                    title="It Takes Two"
                        />
                        <hr />
            </div>
            <div
                        className="cardContainer"
            >
                        <Connect(Card)
                                    backdrop={undefined}
                                    id={324982}
                                    rating={undefined}
                                    title="Step Brothers"
                        />
                        <hr />
            </div>
</div>,
        ],
        "className": "sidebar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Recent Movies",
          },
          "ref": null,
          "rendered": "Recent Movies",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="cardContainer"
>
                <Connect(Card)
                                backdrop={undefined}
                                id={344911}
                                rating={undefined}
                                title="It Takes Two"
                />
                <hr />
</div>,
              <div
                className="cardContainer"
>
                <Connect(Card)
                                backdrop={undefined}
                                id={324982}
                                rating={undefined}
                                title="Step Brothers"
                />
                <hr />
</div>,
            ],
            "className": "sidebarCards",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "344911",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Connect(Card)
                    backdrop={undefined}
                    id={344911}
                    rating={undefined}
                    title="It Takes Two"
/>,
                  <hr />,
                ],
                "className": "cardContainer",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "backdrop": undefined,
                    "id": 344911,
                    "rating": undefined,
                    "title": "It Takes Two",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "hr",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "324982",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Connect(Card)
                    backdrop={undefined}
                    id={324982}
                    rating={undefined}
                    title="Step Brothers"
/>,
                  <hr />,
                ],
                "className": "cardContainer",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "backdrop": undefined,
                    "id": 324982,
                    "rating": undefined,
                    "title": "Step Brothers",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "hr",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
